### DISCUSSION TITLE
Best place to save HTML  ?

### DISCUSSION THREAD

**Discussion by MrVibe** [supabase/supabase#20210] [Questions] [answered]
What would be the best way to save HTML, in database or in flat files ?
With the new feature "read replicas" I am inclined towards database but with database the size of the db increases without any relational data advanatge ( no search / sort ).

I am using supabase in a project which is using a pagebuilder that generates HTML.
Currently I am saving the HTML as a plain text file in subase storage. I read that file when anyone visits the site.

**Comment by GaryAustin1** [comment, marked as answer]
You don't mention sizes, but seems bad idea for cost reasons to do that.  Also with storage you would get the potential for browser and cdn caching if that matters for your app.

**Reply by MrVibe** [reply]
The sizes can vary but it is html the sizes do not exceed 20kb. I agree that cost wise it is better to save HTML. Does supabase public storage already come with CDN ?

**Reply by GaryAustin1** [reply]
Yes Supabase storage is behind a CDN.  It is more advanced for Pro users, but even free has it.
